File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        // comparison to last element of array -> {F, F, F, F, F, T, T, T}
        int n = nums.size();
        int l = -1, r = n - 1;
        while (r - l > 1) {
            int mid = (l + r) >> 1;
            if (nums[mid] <= nums.back()) r = mid;
            else l = mid;
        }
        return nums[r];
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int n = nums.length;
        // if (nums[0] < nums[n - 1]) return nums[0];
        // isGreateThanLastNumber(arr[]): [T, T, T, T, F, F]
        int lo = -1, hi = n;
        while (hi > lo + 1) {
            int mid = (lo + hi) >> 1;
            if (nums[mid] > nums[n - 1])
                lo = mid;
            else
                hi = mid;
        }
        return nums[hi];
    }
}